src/GafferImage/ImageWriter.cpp

```diff
@@ -706,10 +711,24 @@ ImageWriter::ImageWriter( const std::string &name )
 	addChild( new ImagePlug( "in" ) );
 	addChild( new StringPlug( "fileName" ) );
 	addChild( new StringPlug( "channels", Gaffer::Plug::In, "*" ) );
+
+	ColorSpacePtr colorSpace = new ColorSpace( "__colorSpace" );
```

Comment:
I guess you could to be flexible but in practice I think this is unlikely to be useful.
ColorSpaces used in stored files are very unlikely to use context dependent ones.
Let me know if you still think it's a good idea to do so, I think it should be trivial.